apps/web/src/hooks/v3/useV3Positions.ts

```diff
@@ -93,72 +94,49 @@ export function useV3TokenIdsByAccount(
   contractAddress?: Address,
   account?: Address | null | undefined,
 ): { tokenIds: bigint[]; loading: boolean } {
-  const { chainId } = useActiveChainId()
-  const {
-    isLoading: balanceLoading,
-    data: accountBalance,
-    refetch: refetchBalance,
-  } = useContractRead({
-    abi: masterChefV3ABI,
-    address: contractAddress as `0x${string}`,
-    args: [account ?? undefined],
+  const { result: accountBalance, loading: balanceLoading } = useSingleCallResult({
+    contract: useMemo(
+      () =>
+        contractAddress
+          ? {
+              abi: masterChefV3ABI,
+              address: contractAddress as `0x${string}`,
+            }
+          : null,
+      [contractAddress],
+    ),
+    args: useMemo(() => [account ?? undefined] as readonly [`0x${string}`], [account]),
     functionName: 'balanceOf',
-    enabled: !!account && !!contractAddress,
-    watch: true,
-    chainId,
+    options: { enabled: !!account && !!contractAddress },
   })
 
-  const tokenIdsArgs = useMemo(() => {
-    if (accountBalance && account) {
-      const tokenRequests: {
-        abi: typeof masterChefV3ABI
-        address: Address
-        functionName: 'tokenOfOwnerByIndex'
-        args: [Address, number]
-        chainId: number
-      }[] = []
-      for (let i = 0; i < accountBalance; i++) {
-        tokenRequests.push({
-          abi: masterChefV3ABI,
-          address: contractAddress as `0x${string}`,
-          functionName: 'tokenOfOwnerByIndex',
-          args: [account, i],
-          chainId,
-        })
+  const tokenIdCallState = useSingleContractMultipleData({
```

Comment:
I remember it's on purpose to avoid uni multicall, the state return is new ref every fetch

Reply:
Benefits on react query data shallow equals huh